src/main/java/com/github/dockerjava/api/model/PropagationMode.java

```diff
@@ -0,0 +1,49 @@
+package com.github.dockerjava.api.model;
+
+/**
+ * The propagation mode of a file system or file: <code>shared</code>, <code>slave</code> or <code>private</code>.
+ *
+ * @see https://github.com/docker/docker/pull/17034
+ * @since 1.22
+ */
+public enum PropagationMode {
+    /** default */
+    mount_default(""),
```

Comment:
Please let me know if anything is missing. If something about Json is required then please let me know what exactly, ideally with a pointer to other similar code. Would be great if a version with this change could be released soon.